run.py

```diff
@@ -79,20 +53,34 @@ def _run_multiprocess(self, graph):
             results.append(th)
         return results
 
-    def run_locally(self, of_name, update = False):
+    def run_locally(self, output, nworkers, nthreads):
         """Save to file the histograms booked.
 
         Args:
-            of_name (str): Name of the output .root
-                file
+            output (str): Name of the output .root file
+            nworkers (int): number of slaves passed to the
+                multiprocessing.Pool() function
+            nthreads (int): number of threads passed to the
+                EnableImplicitMT function
         """
+        if not isinstance(nthreads, int):
+            raise TypeError('TypeError: wrong type for nthreads')
+        if nthreads < 1:
+            raise ValueError('ValueError: nthreads has to be larger zero')
+        self.nthreads = nthreads
+        if not isinstance(nworkers, int):
+            raise TypeError('TypeError: wrong type for nworkers')
+        if nworkers < 1:
+            raise ValueError('ValueError: nworkers has to be larger zero')
+        logger.info('Computing {} graphs locally using {} workers with {} threads each'.format(
+            len(self.graphs), nworkers, nthreads))
+        pool = Pool(nworkers)
+        final_results = list(pool.map(self._run_multiprocess, self.graphs))
+        final_results = [j for i in final_results for j in i]
         logger.info('Writing {} results to file {}'.format(
-            len(self.final_results), of_name))
-        if update:
-            root_file = TFile(of_name, 'UPDATE')
-        else:
-            root_file = TFile(of_name, 'RECREATE')
-        for op in self.final_results:
+            len(final_results), output))
+        root_file = TFile(output, 'RECREATE')
```

Comment:
are you sure about this? there's no case in which we may need to update a root file instead of overwrite/create a new one?

Reply:
i would always recreate. it's simple! and normally we never update files. it's just too messy to keep track of everything. before, i think we would create a new file and `hadd` stuff together.

Reply:
and it's an option less :)